logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/incremental/EventGatherer.kt

```diff
@@ -133,13 +135,14 @@ internal class EventGathererImpl(
         logger.i("Websocket Open")
         eventRepository
             .pendingEvents()
-            .mapNotNull { offlineEventOrFailure ->
-                when (offlineEventOrFailure) {
-                    is Either.Left -> null
-                    is Either.Right -> offlineEventOrFailure.value
+            .onEach {
+                it.onFailure { failure ->
+                    throw KaliumSyncException("Failure to fetch pending events, aborting Incremental Sync", failure)
```

Comment:
Cool, so as I am understanding, this will bubble up to `SyncExceptionHandler` and be handled by `IncrementalSync` eventually  🤔  ?

Reply:
Yes, exactly.

As we use `flow` extensively in that part of the code, it becomes very spaghetti-like to handle both `Either`, flows of events and `CancellationExceptions`.

So in order to simplify a bit, flows are either emiting or throwing (which can be failure or cancellation).